File: src/main/java/com/example/speedyprintserver/ENTITY/Commande/Commande.java
package com.example.speedyprintserver.ENTITY.Commande;

import com.example.speedyprintserver.ENTITY.Service.*;
import com.example.speedyprintserver.ENTITY.User.Client;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;

@Entity
@Table
@Data
public class Commande {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idCommande ;

    private Integer quantite ;
    private LocalDate dateLivraison ;
    private LocalDate dateCreation ;
    private Boolean origin ; //1 => app ; 0 => in presentie
    private String description ;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Service service ;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Client client ;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private EtatCommande etatCommande ;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private ModeImpression modeImpression;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Dimension dimension;


    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Format format;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Conception conception;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Impression impression;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private TypePapier typePapier;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Faces faces;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Grammage grammage;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Finition finition;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Forme forme;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Couleur couleur;


}

File: speedyprintapp-webapp/src/app/core/Model/Commande/Commande.ts
import { TypePapier } from './../Parametre/TypePapier';
import { Client } from "../Client/Client";
import { Conception } from "../Parametre/Conception";
import { Dimension } from "../Parametre/Dimension";
import { Format } from "../Parametre/Format";
import { Impression } from "../Parametre/Impression";
import { ModeImpression } from "../Parametre/ModeImpression";
import { Service } from "../Parametre/Service";
import { EtatCommande } from "./EtatCommande";
import { Grammage } from '../Parametre/Grammage';
import { Finition } from '../Parametre/Finition';
import { Forme } from '../Parametre/Formes';

export class Commande{
  constructor(){}

  idCommande ?:number ;
  quantite !:number ;
  dateLivraison !:string ;
  dateCreation !:string ;
  origin!:boolean;
  description !:string;

  service !:Service;
  client !:Client;
  etatCommande !:EtatCommande;
  modeImpression !:ModeImpression;
  dimension!:Dimension;
  format!:Format;
  conception!:Conception;
  impression ! :Impression;
  typePapier ?:TypePapier;
  grammage ? : Grammage;
  finition ?:Finition ;
  forme ? : Forme ;
}

const commande : Commande = new Commande();
